apps/src/applab/dataBrowser/ColumnHeader.jsx

```diff
@@ -0,0 +1,157 @@
+/**
+ * @overview Component for adding a new column to the specified table.
+ */
+
+import Radium from 'radium';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import * as dataStyles from './dataStyles';
+
+const styles = {
+  menu: {
+    float: 'right'
+  },
+  icon: {
+    color: 'white'
+  }
+};
+
+const ColumnHeader = React.createClass({
+  propTypes: {
+    columnName: React.PropTypes.string.isRequired,
+    columnNames: React.PropTypes.array.isRequired,
+    deleteColumn: React.PropTypes.func.isRequired,
+    editColumn: React.PropTypes.func.isRequired,
+    isEditable: React.PropTypes.bool.isRequired,
+    isEditing: React.PropTypes.bool.isRequired,
+    renameColumn: React.PropTypes.func.isRequired,
+  },
+
+  getInitialState() {
+    return {
+      newName: this.props.columnName,
+      hasEnteredText: false,
+    };
+  },
+
+  componentDidMount() {
+    if (this.props.isEditing) {
+      ReactDOM.findDOMNode(this.refs.input).select();
```

Comment:
Because `this.refs.input` is a DOM component and not a composite component, `ReactDOM.findDOMNode` isn't necessary here ([as of React 0.14.0](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#dom-node-refs)).

Also, [string refs are considered legacy](https://facebook.github.io/react/docs/more-about-refs.html#the-ref-string-attribute); our styleguide hasn't deprecated them yet, but did you consider callback refs?